hdc/algo/accessors.py

```diff
@@ -480,7 +480,7 @@ def spi(
                 raise ValueError(
                     "Calibration stop cannot be smaller than first timestamp!"
                 )
-            calstop_ix = tix.get_loc(calstop, method="ffill") + 1
+            (calstop_ix,) = tix.get_indexer([calstop], method="ffill") + 1
```

Comment:
this `+ 1` still works like that?
whats the logic behind `bfill` vs `ffill`?

Reply:
`+ 1` is needed for the SPI function downstream where this is not used as an index, but as an array size.

`bfill` / `ffill` is used to select an appropriate date if no exact matching selection is provided. `bfill` is used for the start so if no match, the next date is selected (also clipping the index to 0). `ffill` is used so the previous date is selected to not include an extra date.